### ISSUE TITLE
Bug in markdown parsing

### ISSUE THREAD

**Issue by kytrinyx** [exercism/exercism.io#2697]
This one is tricky.

See `app/views/site/help.erb` line 11. `docs.firewall` is commented out, because our markdown parser botches it. The markdown file in question is `x/docs/md/help/firewall.md`.

When this is included in the page, it eats the actual code section, leaving it empty.
If you write a test for parsing just that file, however, then it works just fine.

**Comment by kytrinyx** [comment]
@mfrost if you have time to take a look, would you give this one a whirl? You might see something obvious that I've missed.

**Comment by mfrost** [comment]
Yep, I'll check this out. Happy new year, btw!

**Comment by kytrinyx** [comment]
Happy new year :)

**Comment by mfrost** [comment]
So as far as I can tell, the parser is choking on the second code block.

```bash
[Environment]::SetEnvironmentVariable("http_proxy","http://[user]:[pass]@[proxy]:[port]")
```

It seems to be swallowing up the contents at some point before `ExercismLib::Markdown::block_code`. It does look a lot like a markdown link...

**Comment by mfrost** [comment]
Maybe something in [this section](https://github.com/vmg/redcarpet/blob/master/ext/redcarpet/markdown.c#L2837-L2863) is failing? The C file is pretty long and hard to follow. I do know that if I add 4 spaces to the front of the line, [is_ref()](https://github.com/vmg/redcarpet/blob/master/ext/redcarpet/markdown.c#L2603-L2604) seems to ignore the powershell code and it comes out correctly.  Stumped otherwise...

**Comment by mfrost** [comment]
Updating to the latest version of redcarpet (3.3.4) seems to fix this issue. Can you confirm @kytrinyx? Here is a handy link for comparison: https://github.com/vmg/redcarpet/compare/v3.2.3...v3.3.4 

Perhaps it was https://github.com/vmg/redcarpet/pull/435?

**Comment by kytrinyx** [comment]
I think you're right that it has something to do with the second code block looking suspiciously like a link. I'll update redcarpet and confirm the fix.

**Comment by kytrinyx** [comment]
Thanks @mfrost that did it.

**Comment by mfrost** [comment]
Booyah. :+1:
